drivers/pwm/pwm_stm32.c

```diff
@@ -293,6 +294,20 @@ static int pwm_stm32_set_cycles(const struct device *dev, uint32_t channel,
 	set_timer_compare[channel - 1u](timer, pulse_cycles);
 	LL_TIM_SetAutoReload(timer, period_cycles);
 
+#if defined(IS_TIM_BREAK_INSTANCE)
```

Comment:
Late catch:

Please move this piece of code to `pwm_stm32_init()`. Since value is coming from device tree, it should be configured at boot time and not on `pwm_set_cycles()` call by application.
This is also how other DT parameters are handle.